fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1739,14 +1729,19 @@ private ExtractedSearchParameters extractSearchParameters(Resource fhirResource,
                 // Augment the extracted parameter list with special values we use to represent compartment relationships.
                 // These references are stored as tokens and are used by the search query builder
                 // for compartment-based searches
-                addCompartmentParams(allParameters, fhirResource);
+                addCompartmentParams(allParameters, fhirResource.getClass().getSimpleName());
 
                 // If this is a definitional resource, augment the extracted parameter list with a composite
                 // parameter that will be used for canonical searches. It will contain the url and version
                 // values.
                 addCanonicalCompositeParam(allParameters);
             }
 
+            // Remove parameters that aren't to be stored
+            boolean anyRemoved = allParameters.removeIf(value -> !value.isForStoring());
+            if (anyRemoved) {
+                log.warning("Removed unexpectedly found extracted search parameter values that weren't for storing");
```

Comment:
warning text is a little confusing

Reply:
we should never hit it anyway.  open to suggestion

Reply:
is this better?
```suggestion
                log.warning("The set of extracted parameters values unexpectedly contained values that weren't for storing; these have been removed");
```